server/src/main/java/org/apache/cassandra/sidecar/modules/CoordinationModule.java

```diff
@@ -70,6 +72,18 @@ ElectorateMembership electorateMembership(InstanceMetadataFetcher instanceMetada
                                               CQLSessionProvider cqlSessionProvider,
                                               SidecarConfiguration configuration)
     {
-        return new MostReplicatedKeyspaceTokenZeroElectorateMembership(instanceMetadataFetcher, cqlSessionProvider, configuration);
+        String strategy = configuration.serviceConfiguration()
+                                       .coordinationConfiguration()
+                                       .clusterLeaseClaimConfiguration()
+                                       .electorateMembershipStrategy();
+        switch (strategy)
+        {
+            case "MostReplicatedKeyspaceTokenZeroElectorateMembership":
+                return new MostReplicatedKeyspaceTokenZeroElectorateMembership(instanceMetadataFetcher, cqlSessionProvider, configuration);
+            case "SidecarInternalTokenZeroElectorateMembership":
+                return new SidecarInternalTokenZeroElectorateMembership(instanceMetadataFetcher, configuration);
+            default:
+                throw new ConfigurationException("Invalid electorate membership strategy value '" + strategy + "'");
+        }
```

Comment:
can we move this code to a factory class? Module code is hard to testable, and complex code should be tested.

Reply:
done